comp/core/secrets/impl/fetch_secret.go

```diff
@@ -111,15 +110,9 @@ func (r *secretResolver) fetchSecretBackendVersion() (string, error) {
 		return r.versionHookFunc()
 	}
 
-	// check if binary likely supports --version
-	content, err := os.ReadFile(r.backendCommand)
-	if err != nil {
-		log.Debugf("Could not read binary %s to check for --version support: %v", r.backendCommand, err)
-		return "", fmt.Errorf("could not read binary to check version support")
-	}
-
-	if !bytes.Contains(content, []byte("--version")) {
-		return "", fmt.Errorf("binary does not appear to support --version")
+	// Only get version when secret_backend_type is used
+	if r.backendType == "" {
```

Comment:
Could you add a test for this case? It should be similar to `TestFetchSecretBackendVersionTimeout` except (1) don't assign to `versionHookFunc` and (2) use `assert.ErrorContains()` to check that the error message matches something relevant, like that it contains "secret_backend_type" in its message.